code/modules/mapping/space_management/space_transition.dm

```diff
@@ -60,39 +60,24 @@
 /datum/controller/subsystem/mapping/proc/setup_map_transitions() //listamania
 	var/list/transition_levels = list()
 	var/list/cached_z_list = z_list
+	var/linkage
 	for(var/datum/space_level/level as anything in cached_z_list)
-		if (level.linkage == CROSSLINKED)
-			transition_levels.Add(level)
-
-	var/grid_diameter = (length(transition_levels) * 2) + 1
-	var/list/grid = new /list(grid_diameter ** 2)
+		if(level.linkage != CROSSLINKED && level.linkage != GRIDLINKED)
+			continue
 
-	var/datum/space_transition_point/point
-	for(var/x in 1 to grid_diameter)
-		for(var/y in 1 to grid_diameter)
-			point = new/datum/space_transition_point(x, y, grid)
-			grid[CHORDS_TO_1D(x, y, grid_diameter)] = point
-	for(point as anything in grid)
-		point.set_neigbours(grid, grid_diameter)
+		transition_levels.Add(level)
 
-	var/center = round(grid_diameter / 2)
-	point = grid[CHORDS_TO_1D(grid_diameter, center, center)]
-	grid.Cut()
+		if(linkage)
+			if(level.linkage != linkage)
+				// Either you be gridlinked or crosslinked, both is uhhh... a headache
+				stack_trace("Mixed linkages detected in z-level neighbour transitions!")
+		else
+			linkage = level.linkage
```

Comment:
```suggestion
		if(!isnull(linkage) && level.linkage != linkage)
			// Either you be gridlinked or crosslinked, both is uhhh... a headache
			stack_trace("Mixed linkages detected in z-level neighbour transitions!")
			continue
		linkage = level.linkage
```
confused me reading it, didn't realize it was a guard case.
also, should we say which level disagrees? to help mappers